### PR TITLE
Rename and narrow rule `no-down-event-binding` to `no-pointer-down-event-binding`

### PR DESCRIPTION
**Pull request by jfdnc**
Recommended rule `no-down-event-binding` references [WCAG failure of success criteria](https://www.w3.org/WAI/WCAG21/Techniques/failures/F101) for `down` events that only pertain to pointer-down events. Current implementation of `no-down-event-binding` also applies to `keydown` events, which does not fit the intention of the WCAG docs, and therefore should not be included in this lint rule.

- updates related file names to reflect `no-pointer-down-event-binding`
- updates node check for inclusion in "mousedown"|"pointerdown"
- updates related doc and test files as appropriate

### CODE REVIEW

**Comment by jfdnc** [comment]
Just a ping here as this PR and the [issue for it](https://github.com/ember-template-lint/ember-template-lint/issues/2028) aren't being acknowledged. This can be moved forward or not, I just want to be sure that the maintainers here are aware of the apparently incorrect application of this rule preference. Any thoughts or opinions are appreciated.

File: lib/config/a11y.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ module.exports = {
     'no-abstract-roles': 'error',
     'no-accesskey-attribute': 'error',
     'no-aria-hidden-body': 'error',
-    'no-down-event-binding': 'error',
+    'no-pointer-down-event-binding': 'error',
```
Comment (bertdeblock) [inline lib/config/a11y.js:9]: Should probably move down to respect the order of this list. Same goes for the `recommended` config file.

File: lib/rules/no-pointer-down-event-binding.js [hunk 2]
```diff
@@ -26,20 +27,20 @@ function isActionModifer(node) {
...
 /**
- * Check if an event name is a "down" event
+ * Check if an event name is a pointer "down" event
  *
  * @param {string} eventName
  * @return {boolean}
  */
-function isDownEvent(eventName) {
-  return eventName.toLowerCase().endsWith('down');
+function isPointerDownEvent(eventName) {
```
Comment (bertdeblock) [inline lib/rules/no-pointer-down-event-binding.js:35]: I think it's best that this util still returns a boolean like it used to, now it returns either an array or `null`.

File: test/unit/rules/no-pointer-down-event-binding-test.js [hunk 3]
```diff
@@ -1,61 +1,61 @@
...
+const { ERROR_MESSAGE } = require('../../../lib/rules/no-pointer-down-event-binding');
 const generateRuleTests = require('../../helpers/rule-test-harness');
 
 generateRuleTests({
-  name: 'no-down-event-binding',
+  name: 'no-pointer-down-event-binding',
 
   config: true,
 
   good: [
```
Comment (bertdeblock) [inline test/unit/rules/no-pointer-down-event-binding-test.js:11]: Personally, I would keep the old good cases as they are still allowed and to make sure not to introduce any regressions.

**Review by bertdeblock — COMMENTED** [review]
👋 Left a few suggestions.

Also wanted to point out that, even though the rename makes sense, this might break people's lint build if they use the old name in disable statements or for turning off the rule completely via the config file. So I think we should either keep the old rule name and rename later on _or_ mark this PR as breaking?

File: lib/config/a11y.js (hunk 1 above)
Comment (jfdnc) [reply lib/config/a11y.js:9]: Good catch, will alpha sort this

**Comment by jfdnc** [comment]
> So I think we should either keep the old rule name and rename later on _or_ mark this PR as breaking?

@bertdeblock I think we should prefer to mark this as breaking. The original intent of introducing the `no-down-event-binding` rule was to address the issues with down-events pointed out [here](https://www.w3.org/WAI/WCAG21/Techniques/failures/F101). Maintaining both that rule and the one is unnecessarily over-specified, and `no-down-event-binding` would continue to block `key` events for no purpose.

Thanks for the feedback here, will make the suggested tweaks!

File: test/unit/rules/no-pointer-down-event-binding-test.js (hunk 3 above)
Comment (jfdnc) [reply test/unit/rules/no-pointer-down-event-binding-test.js:11]: I think if this pointer-down linting concern had been introduced first, we wouldn't have had a reason to include `key` events in the scope of the tests, so it made sense to me to remove those. I don't think anything is really gained by having those included. I can add those back in the "good" checks if you feel strongly on that point, though.

Comment (bertdeblock) [reply test/unit/rules/no-pointer-down-event-binding-test.js:11]: Nope, no strong feelings on this one. I'm just a contributor btw, so not my decision to make. But I've seen this mentioned before in PRs for this project, that's why I mentioned it.

**Comment by bmish** [comment]
Just FYI, the v4 release (https://github.com/ember-template-lint/ember-template-lint/issues/1908) is coming up in the next week or two. So if we want to make breaking changes, now is the time to do it. CC: @elwayman02

**Comment by drewlee** [comment]
@jfdnc I'm inquiring to see if these changes can be picked back up again or maybe modified so they can be released as a non-breaking change. (Maybe make the key down event a warning instead of an error?) Would be happy to assist in any way. Thanks!
CC @bmish

**Comment by bmish** [comment]
Usually it's best to use an opt-in rule option to avoid a breaking change. But note that we will have another major release coming up (#2319) if breaking changes are desired.

**Comment by drewlee** [comment]
@bmish I'd like to have my team pick this up, as this PR seems to have gone stale and abandoned. The simplest path sounds like a breaking change that can be rolled into the v5 release. We will likely address via a new PR. Since I don't have insight into the exact date v5 is slated for, I just wanted to make sure we have enough time to land this. Thanks!

**Comment by judithhinlung** [comment]
I think this refactor could be ready to merge with a few changes. I have updated this PR to pull in the changes from master since its last commit, and I added `onmousedown` and `onpointerdown` to the list of pointer-down events. Since this is a breaking change, I have also removed this rule from the list of recommended rules.

File: lib/config/3-x-recommended.js [hunk 4]
```diff
@@ -43,6 +42,7 @@ export default {
     'no-outlet-outside-routes': 'error',
     'no-partial': 'error',
     'no-passed-in-event-handlers': 'error',
+    'no-pointer-down-event-binding': 'error',
```
Comment (drewlee) [inline lib/config/3-x-recommended.js:45]: I believe the `3-x-recommended.js` config was specific to version 3.0 and is no longer in use/relevant, so this change isn't necessary.

**Comment by judithhinlung** [comment]
Due to the name change, leaving the rule name in 3-x-recommended.js as `no-down-event-binding` will cause the following test failure:
 FAIL  test/unit/default-config-test.js
  ● default configurations › 3-x-recommended.js › should contain only valid rules

**Comment by drewlee** [comment]
@bmish Can we get some eyes on this and make sure we're on the right track? Thanks!

File: lib/rules/no-pointer-down-event-binding.js [hunk 5]
```diff
@@ -19,25 +20,27 @@ function isOnModifier(node) {
...
+  return ['mousedown', 'onmousedown', 'pointerdown', 'onpointerdown'].includes(
```
Comment (bmish) [inline lib/rules/no-pointer-down-event-binding.js:34]: Can you declare this array in a `const` outside the function so it doesn't get recreated each time the function is called?

**Review by bmish — APPROVED** [review]
Looks ready to go as part of the next major release #2319.

File: lib/rules/no-pointer-down-event-binding.js (hunk 5 above)
Comment (judithhinlung) [reply lib/rules/no-pointer-down-event-binding.js]: Yes, I have updated the rule to declare the pointer-down event bindings in a set called during rule setup.

File: README.md [hunk 6]
```diff
@@ -152,6 +151,8 @@ Each rule has emojis denoting:
 | [no-outlet-outside-routes](./docs/rule/no-outlet-outside-routes.md)                                       | ✅  |     |     |     |
 | [no-partial](./docs/rule/no-partial.md)                                                                   | ✅  |     |     |     |
 | [no-passed-in-event-handlers](./docs/rule/no-passed-in-event-handlers.md)                                 | ✅  |     |     |     |
+<<<<<<< HEAD
```
Comment (bertdeblock) [inline README.md:154]: 😬

File: docs/rule/no-action.md [hunk 7]
```diff
@@ -4,7 +4,7 @@
 
 What's wrong with `{{action}}`?
 
-"Action" is an overloaded term in Ember parlance. Actions are:
```
Comment (drewlee) [inline docs/rule/no-action.md:7]: @judithhinlung It looks like you got a bad merge as there are a bunch of unrelated changes in this PR now.

**Comment by judithhinlung** [comment]
I have reverted the previous rebase to merge with master again, and the previous issues should be resolved.

**Comment by jfdnc** [comment]
Been out of the loop with OS things for a while, @judithhinlung thanks so much for picking this up! You rock 🎉
